samples/bluetooth/broadcast_audio_assistant/src/main.c

```diff
@@ -0,0 +1,469 @@
+	struct bt_uuid_16 adv_uuid;
+
+	switch (data->type) {
+	case BT_DATA_NAME_SHORTENED:
+	case BT_DATA_NAME_COMPLETE:
+		memcpy(sr_info->bt_name, data->data, MIN(data->data_len, NAME_LEN - 1));
+		return true;
+	case BT_DATA_BROADCAST_NAME:
+		memcpy(sr_info->broadcast_name, data->data, MIN(data->data_len, NAME_LEN - 1));
+		return true;
+	case BT_DATA_SVC_DATA16:
+		/* Check for Broadcast ID */
+		if (data->data_len < BT_UUID_SIZE_16 + BT_AUDIO_BROADCAST_ID_SIZE) {
+			return true;
+		}
+
+		if (!bt_uuid_create(&adv_uuid.uuid, data->data, BT_UUID_SIZE_16)) {
+			return true;
+		}
+
+		if (bt_uuid_cmp(&adv_uuid.uuid, BT_UUID_BROADCAST_AUDIO) != 0) {
+			return true;
+		}
+
+		sr_info->broadcast_id = sys_get_le24(data->data + BT_UUID_SIZE_16);
+		return true;
+	case BT_DATA_UUID16_SOME:
+	case BT_DATA_UUID16_ALL:
+		/* Check for BASS and PACS */
+		if (data->data_len % sizeof(uint16_t) != 0U) {
+			printk("UUID16 AD malformed\n");
+			return true;
+		}
+
+		for (size_t i = 0; i < data->data_len; i += sizeof(uint16_t)) {
+			const struct bt_uuid *uuid;
+			uint16_t u16;
+
+			memcpy(&u16, &data->data[i], sizeof(u16));
+			uuid = BT_UUID_DECLARE_16(sys_le16_to_cpu(u16));
+
+			if (bt_uuid_cmp(uuid, BT_UUID_BASS)) {
+				sr_info->has_bass = true;
+				continue;
+			}
+
+			if (bt_uuid_cmp(uuid, BT_UUID_PACS)) {
+				sr_info->has_pacs = true;
+				continue;
+			}
+		}
+		return true;
+	default:
+		return true;
+	}
+}
+
+static bool is_substring(const char *substr, const char *str)
+{
+	const size_t str_len = strlen(str);
+	const size_t sub_str_len = strlen(substr);
+
+	if (sub_str_len > str_len) {
+		return false;
+	}
+
+	for (size_t pos = 0; pos < str_len; pos++) {
+		if (pos + sub_str_len > str_len) {
+			return false;
+		}
+
+		if (strncasecmp(substr, &str[pos], sub_str_len) == 0) {
+			return true;
+		}
+	}
+
+	return false;
+}
+
+static void scan_recv_cb(const struct bt_le_scan_recv_info *info,
+			 struct net_buf_simple *ad)
+{
+	int err;
+	struct scan_recv_info sr_info = {0};
+
+	if (scanning_for_broadcast_source) {
+		/* Scan for and select Broadcast Source */
+
+		sr_info.broadcast_id = INVALID_BROADCAST_ID;
+
+		/* We are only interested in non-connectable periodic advertisers */
+		if ((info->adv_props & BT_GAP_ADV_PROP_CONNECTABLE) != 0 ||
+		    info->interval == 0) {
+			return;
+		}
+
+		bt_data_parse(ad, device_found, (void *)&sr_info);
+
+		if (sr_info.broadcast_id != INVALID_BROADCAST_ID) {
+			printk("Broadcast Source Found:\n");
+			printk("  BT Name:        %s\n", sr_info.bt_name);
+			printk("  Broadcast Name: %s\n", sr_info.broadcast_name);
+			printk("  Broadcast ID:   0x%06x\n\n", sr_info.broadcast_id);
+
+			if (strlen(CONFIG_SELECT_SOURCE_NAME) > 0U) {
+				/* Compare names with CONFIG_SELECT_SOURCE_NAME */
+				if (is_substring(CONFIG_SELECT_SOURCE_NAME, sr_info.bt_name) ||
+				    is_substring(CONFIG_SELECT_SOURCE_NAME,
+				    sr_info.broadcast_name)) {
```

Comment:
Minor: the split line is still not correctly indented. You're matching the `if (` when you should instead be matching `is_substring(`